sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobAsyncClientBase.java

```diff
@@ -1605,4 +1625,292 @@ public String generateSas(BlobServiceSasSignatureValues blobServiceSasSignatureV
+     * @param logger {@link ClientLogger}
+     * @return {@link QuickQuerySerialization}
+     */
+    private static QuickQuerySerialization transformSerialization(BlobQuerySerialization userSerialization,
+        ClientLogger logger) {
+        if (userSerialization == null) {
+            return null;
+        }
+
+        QuickQueryFormat generatedFormat = new QuickQueryFormat();
+        if (userSerialization instanceof BlobQueryDelimitedSerialization) {
+
+            generatedFormat.setType(QuickQueryFormatType.DELIMITED);
+            generatedFormat.setDelimitedTextConfiguration(transformDelimited(
+                (BlobQueryDelimitedSerialization) userSerialization));
+
+        } else if (userSerialization instanceof BlobQueryJsonSerialization) {
+
+            generatedFormat.setType(QuickQueryFormatType.JSON);
+            generatedFormat.setJsonTextConfiguration(transformJson(
+                (BlobQueryJsonSerialization) userSerialization));
+
+        } else {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                String.format("'input' must be one of %s or %s", BlobQueryJsonSerialization.class.getSimpleName(),
+                    BlobQueryDelimitedSerialization.class.getSimpleName())));
+        }
+        return new QuickQuerySerialization().setFormat(generatedFormat);
+    }
+
+    /**
+     * Transforms a BlobQuickQueryDelimitedSerialization into a DelimitedTextConfiguration.
+     *
+     * @param delimitedSerialization {@link BlobQueryDelimitedSerialization}
+     * @return {@link DelimitedTextConfiguration}
+     */
+    private static DelimitedTextConfiguration transformDelimited(
+        BlobQueryDelimitedSerialization delimitedSerialization) {
+        if (delimitedSerialization == null) {
+            return null;
+        }
+        return new DelimitedTextConfiguration()
+            .setColumnSeparator(charToString(delimitedSerialization.getColumnSeparator()))
+            .setEscapeChar(charToString(delimitedSerialization.getEscapeChar()))
+            .setFieldQuote(charToString(delimitedSerialization.getFieldQuote()))
+            .setHeadersPresent(delimitedSerialization.isHeadersPresent())
+            .setRecordSeparator(charToString(delimitedSerialization.getRecordSeparator()));
+    }
+
+    /**
+     * Transforms a BlobQuickQueryJsonSerialization into a JsonTextConfiguration.
+     *
+     * @param jsonSerialization {@link BlobQueryJsonSerialization}
+     * @return {@link JsonTextConfiguration}
+     */
+    private static JsonTextConfiguration transformJson(BlobQueryJsonSerialization jsonSerialization) {
+        if (jsonSerialization == null) {
+            return null;
+        }
+        return new JsonTextConfiguration()
+            .setRecordSeparator(charToString(jsonSerialization.getRecordSeparator()));
+    }
+
+    private static String charToString(char c) {
+        return c == '\0' ? "" : Character.toString(c);
```

Comment:
Why do we treat \0 specially? Can customer choose \0 as record separator (for whatever weird reason)?

Reply:
I thought that Karim told me that null should be "" but I have sent an email reconfirming.

Reply:
null is the default param for field quote and escape char according to the QQ folks.